File: Problem_4/main.js
function isPalindrome(number)
{
    let reversed = number.toString().split("").reverse().join("");
    return number == reversed;
}


let digits = 3;
palindromes_arr = [];
for (let i = Math.pow(10, digits - 1); i < Math.pow(10, digits); i++)
{
    for (let j = i; j < Math.pow(10, digits); j++)
    {
        let prod = i * j;
        if (isPalindrome(prod))
        {
            palindromes_arr.push(prod);
        }
    }
}

console.log(Math.max(...palindromes_arr));

File: Problem_4/main.py
def isPalindrome(x):
    return str(x) == str(x)[::-1]

max_product = 1
for i in range(100, 1000):
    for j in range(i, 1000):
        if isPalindrome(i * j) and (i * j > max_product):
            max_product = i * j

print(max_product)